### PR TITLE
[sfmDataIO] sample scene generation for I/O version check

### PR DESCRIPTION
**Pull request by servantf**
For backward compatibility reason, we have to make sure old scene files are correctly imported in new versions.
This code is meant to generate scene files. They will be used to validate the correct importation in unit tests.

### CODE REVIEW

File: src/aliceVision/sfmDataIO/sceneSample.cpp [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            }
+        }
+    }
+
+    int w = 4092;
+    int h = 2048;
+    double focalLengthPixX = 1000.0;
+    double focalLengthPixY = 2000.0;
+    double ppx = 2020.0;
+    double ppy = 1040.0;
```
Comment (simogasp) [inline src/aliceVision/sfmDataIO/sceneSample.cpp:33]:
```suggestion
    const int w = 4092;
    const int h = 2048;
    const const double focalLengthPixX = 1000.0;
    const double focalLengthPixY = 2000.0;
    const const double ppx = 2020.0;
    const double ppy = 1040.0;
```

File: src/aliceVision/sfmDataIO/sceneSample.cpp [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+bool generateSampleScene(sfmData::SfMData & output)
+{
+    // Generate points on a cube
+    IndexT idpt = 0;
+    for(int x = -10; x <= 10; x++)
+    {
+        for(int y = -10; y <= 10; y++)
+        {
+            for(int z = -10; z <= 10; z++)
```
Comment (simogasp) [inline src/aliceVision/sfmDataIO/sceneSample.cpp:20]:
```suggestion
    for(int x = -10; x <= 10; ++x)
    {
        for(int y = -10; y <= 10; ++y)
        {
            for(int z = -10; z <= 10; ++z)
```

File: src/aliceVision/sfmDataIO/sceneSample.cpp [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    output.getIntrinsics().emplace(1, std::make_shared<camera::EquiDistantRadialK3>(w, h, focalLengthPixX, ppx, ppy, 0.1, 0.05, -0.001));
+
+    // Generate poses on another cube
+    IndexT idpose = 0;
+    IndexT idview = 0;
+    for(int x = -1; x <= 1; x++)
+    {
+        for(int y = -1; y <= 1; y++)
+        {
+            for(int z = -1; z <= 1; z++)
```
Comment (simogasp) [inline src/aliceVision/sfmDataIO/sceneSample.cpp:45]:
```suggestion
    for(int x = -1; x <= 1; ++x)
    {
        for(int y = -1; y <= 1; ++y)
        {
            for(int z = -1; z <= 1; ++z)
```

File: src/aliceVision/sfmDataIO/sceneSample.cpp [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+            {
+                Eigen::Vector3d thetau({x, y, z});
```
Comment (simogasp) [inline src/aliceVision/sfmDataIO/sceneSample.cpp:47]:
```suggestion
                const Eigen::Vector3d thetau({x, y, z});
```

File: src/aliceVision/sfmDataIO/sceneSample.cpp [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+                Eigen::AngleAxis<double> aa(thetau.norm(), thetau.normalized());
```
Comment (simogasp) [inline src/aliceVision/sfmDataIO/sceneSample.cpp:48]:
```suggestion
                const Eigen::AngleAxis<double> aa(thetau.norm(), thetau.normalized());
```

File: src/aliceVision/sfmDataIO/sceneSample.cpp [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+                output.getPoses().emplace(idpose, geometry::Pose3(aa.toRotationMatrix(), Vec3({x,y,z})));
+                idpose++;
```
Comment (simogasp) [inline src/aliceVision/sfmDataIO/sceneSample.cpp:51]:
```suggestion
                ++idpose;
```

File: src/aliceVision/sfmDataIO/sceneSample.cpp [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+                for (auto itIntrinsic : output.getIntrinsics())
+                {
+                    output.getViews().emplace(idview, std::make_shared<sfmData::View>("", idview, itIntrinsic.first, idpose, w, h));
+                    idview++;
```
Comment (simogasp) [inline src/aliceVision/sfmDataIO/sceneSample.cpp:56]:
```suggestion
                    ++idview;
```

File: src/aliceVision/sfmDataIO/sceneSample.cpp [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+        {
+            for(int z = -1; z <= 1; z++)
+            {
+                Eigen::Vector3d thetau({x, y, z});
+                Eigen::AngleAxis<double> aa(thetau.norm(), thetau.normalized());
+
+                output.getPoses().emplace(idpose, geometry::Pose3(aa.toRotationMatrix(), Vec3({x,y,z})));
+                idpose++;
+
+                for (auto itIntrinsic : output.getIntrinsics())
```
Comment (simogasp) [inline src/aliceVision/sfmDataIO/sceneSample.cpp:53]:
```suggestion
                for (const auto itIntrinsic : output.getIntrinsics())
```

File: src/aliceVision/sfmDataIO/sceneSample.hpp [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include <aliceVision/sfmData/SfMData.hpp>
+
+namespace aliceVision {
+namespace sfmDataIO {
+
+bool generateSampleScene(sfmData::SfMData & output);
```
Comment (simogasp) [inline src/aliceVision/sfmDataIO/sceneSample.hpp:14]: please add doxygen explaining what the function does

**Review by simogasp — DISMISSED** [review]

File: src/aliceVision/version.hpp [hunk 10]
```diff
@@ -18,3 +18,23 @@
 #define ALICEVISION_VERSION_STRING ALICEVISION_TO_STRING(ALICEVISION_VERSION_MAJOR) "." \
                              ALICEVISION_TO_STRING(ALICEVISION_VERSION_MINOR) "." \
                              ALICEVISION_TO_STRING(ALICEVISION_VERSION_REVISION)
+
+                             
+
+inline bool isVersionOlder(const std::vector<uint32_t> & reference, const std::vector<uint32_t> & compare)
+{
+    for (int i = 0; i < reference.size(); i++)
```
Comment (simogasp) [inline src/aliceVision/version.hpp:26]:
```suggestion
    for (std::size_t i = 0; i < reference.size(); ++i)
```

File: src/aliceVision/version.hpp [hunk 11]
```diff
@@ -18,3 +18,23 @@
 #define ALICEVISION_VERSION_STRING ALICEVISION_TO_STRING(ALICEVISION_VERSION_MAJOR) "." \
                              ALICEVISION_TO_STRING(ALICEVISION_VERSION_MINOR) "." \
                              ALICEVISION_TO_STRING(ALICEVISION_VERSION_REVISION)
+
+                             
+
+inline bool isVersionOlder(const std::vector<uint32_t> & reference, const std::vector<uint32_t> & compare)
```
Comment (fabiencastan) [inline src/aliceVision/version.hpp:24]: seems unused:
version is a string, a define or a Vec3, but I do not see the case with std::vector.

**Review by fabiencastan — APPROVED** [review]

**Pull request merged**